org.kde.optiimage.json

```diff
@@ -0,0 +1,93 @@
+{
+    "id": "org.kde.optiimage",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "6.8",
+    "sdk": "org.kde.Sdk",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "optiimage",
+    "finish-args": [
+        "--filesystem=host",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri"
+    ],
+    "modules": [
+        {
+            "name": "jpegoptim",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/tjko/jpegoptim/archive/refs/tags/v1.5.5.tar.gz",
+                    "sha256": "90a309d1c092de358bb411d702281ac3039b489d03adb0bc3c4ef04cf0067d38"
+                }
+            ]
+        },
+        {
+            "name": "oxipng",
+            "buildsystem": "simple",
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml --verbose",
+                "cargo --offline build --release --verbose",
+                "install -Dm 755 target/release/oxipng -t /app/bin"
+            ],
+            "build-options": {
+                "append-path": "/usr/lib/sdk/rust-stable/bin",
+                "env": {
+                    "CARGO_HOME": "/run/build/oxipng/cargo"
+                }
+            },
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/shssoichiro/oxipng/archive/refs/tags/v9.1.3.tar.gz",
```

Comment:
according to https://github.com/shssoichiro/oxipng/issues/518 they support aarch64, so might want to report it there.

Reply:
i see at least one other app building 9.1.1 on aarch64

Reply:
yeah here is the update to 9.1.1 for the nighlty flatpak build https://invent.kde.org/graphics/optiimage/-/merge_requests/13

Reply:
I could update the PR to use oxipng 9.1.1 instead of 9.1.3, to match that upstream MR, but I'm not sure it would help.  The error was `error: linker 'aarch64-linux-gnu-gcc' not found`, which would be an issue with the build environment, not the oxipng source, right?

Reply:
no

Reply:
Ah, I see.  They did make a change in oxipng 9.1.3 that might be related: https://github.com/shssoichiro/oxipng/commit/e2d58c112ba8041f47dd94e5d130b142208562f8

I'll change this PR to use 9.1.1 like upstream OptiImage.

Reply:
> according to [shssoichiro/oxipng#518](https://github.com/shssoichiro/oxipng/issues/518) they support aarch64, so might want to report it there.

If you read the MR, it's just wrong. They don't support aarch64. They support cross-compiling for it.

https://github.com/shssoichiro/oxipng/pull/534/files#diff-9a4f3e4537ebb7474452d131b0d969d89a51286f4269aac5ef268e712be17268R1

```
[📦 org.freedesktop.Sdk ~]$ which aarch64-linux-gnu-gcc
/usr/bin/which: no aarch64-linux-gnu-gcc in (/app/bin:/usr/bin)
[📦 org.freedesktop.Sdk ~]$ arch 
aarch64
```

Reply:
then a good idea to ask them to support it natively. The original issue was talking about native aarch64

Reply:
it was copiloted.

(ie Microsoft wasting resource on stuff no one asked for, instead of fixing stuff people asked for like aarch64 runners)

Reply:
It looks like we're having the same error after downgrading to oxipng 9.1.1 (see https://buildbot.flathub.org/#/builders/31/builds/15654/steps/7/logs/stdio ) so if the workaround from https://github.com/flathub/flathub/pull/5930#issuecomment-2542723538 works, should I update back to the latest oxipng release, 9.1.3?

Reply:
Yes if the workaround works it should be fine with 9.1.3 too.